src/main/java/com/github/cameltooling/lsp/internal/util/MavenUtils.java

```diff
@@ -0,0 +1,176 @@
+			try {
+				Model model = getModelFromPOM(pom);
+				if (model.getDependencies() != null) {
+					Dependency dep = getAnyCamelDependencyFromDependencies(model);
+					if (dep == null && model.getDependencyManagement() != null) {
+						dep = getAnyCamelDependencyFromDependencyManagement(model);
+					}
+					if (dep != null) {
+						return dep.getVersion();
+					}
+				}
+			} catch (XmlPullParserException | IOException ex) {
+				LOGGER.error("Error retrieving the Camel version from the project.", ex);
+			}
+		}
+		return fallBackCamelVersion;
+	}
+
+	public static boolean isDependencyConfiguredInPom(Model model, Dependency dependency) {
+		return getDependencyFromModel(model, dependency) != null;
+	}
+	
+	public static Dependency getDependencyFromModel(Model model, Dependency dependency) {
+		Dependency dep = getDependencyFromDependenciesSection(model, dependency);
+		if (dep == null) {
+			dep = getDependencyFromDependencyManagementSection(model, dependency);
+		}
+		return dep;
+	}
+	
+	public static Dependency getDependencyFromDependenciesSection(Model model, Dependency dependency) {
+		if (model != null && model.getDependencies() != null) {
+			for (Dependency d : model.getDependencies()) {
+				if (d.getGroupId().equals(dependency.getGroupId()) &&
+					d.getArtifactId().equals(dependency.getArtifactId())) {
+					return d;
+				}
+			}
+		}
+		return null;
+	}
+	
+	public static Dependency getDependencyFromDependencyManagementSection(Model model, Dependency dependency) {
+		if (model != null && model.getDependencyManagement() != null && model.getDependencyManagement().getDependencies() != null) {
+			for (Dependency d : model.getDependencyManagement().getDependencies()) {
+				if (d.getGroupId().equals(dependency.getGroupId()) &&
+					d.getArtifactId().equals(dependency.getArtifactId())) {
+					return d;
+				}
+			}
+		}
+		return null;
+	}
+	
+	public static Dependency getAnyCamelDependencyFromDependencies(Model model) {
+		for (Dependency dep : model.getDependencies()) {
+			if (dep.getGroupId().equals(CAMEL_GID) && dep.getArtifactId().startsWith(CAMEL_AID_PREFIX) && dep.getVersion() != null) {
+				return dep;
+			}
+		}
+		return null;
+	}
+
+	public static Dependency getAnyCamelDependencyFromDependencyManagement(Model model) {
+		for (Dependency dep : model.getDependencyManagement().getDependencies()) {
+			if (dep.getGroupId().equals(CAMEL_GID) && dep.getArtifactId().startsWith(CAMEL_AID_PREFIX) && dep.getVersion() != null) {
+				return dep;
+			}
+		}
+		return null;
+	}
+
+	public static File findPomInReferencedURI(String docUri) {
+		File pom = null;
+		URI uri = URI.create(docUri);
+		File f = new File(uri.getPath());
+		while (f.getParentFile() != null && pom == null) {
+			f = f.getParentFile();
+			pom = getPomFromPath(f);
+		}
+		return pom;
+	}
+
+	public static File getPomFromPath(File path) {
+		File[] files = path.listFiles( (File pathname) -> {
+			return pathname.getName().equalsIgnoreCase("pom.xml");
```

Comment:
better to put "pom.xml" constant on the left of the equalsIgnoreCase to avoid potential NPE